src/pages/networks/forms/NetworkLocalPeeringForm.tsx

```diff
@@ -0,0 +1,220 @@
+          id="customTargetProject"
+          name="customTargetProject"
+          type="text"
+          placeholder="Enter target project name"
+          value={formik.values.customTargetProject}
+          onChange={formik.handleChange}
+          onBlur={formik.handleBlur}
+          disabled={isEditing || !!editRestriction}
+          title={editRestriction}
+        />
+      )}
+
+      {isEditing ? (
+        <Input
+          id="customTargetNetwork"
+          type="text"
+          label="Target network"
+          value={formik.values.targetNetwork}
+          disabled={isEditing}
+          title={"Target network cannot be changed"}
+          required
+        />
+      ) : (
+        formik.values.targetProject &&
+        formik.values.targetProject !== projectOtherLabel && (
+          <NetworkSelector
+            id="targetNetwork"
+            name="targetNetwork"
+            label="Target network"
+            value={formik.values.targetNetwork}
+            setValue={(value) => {
+              formik.setFieldValue("targetNetwork", value, false);
+              if (value !== networkOtherLabel) {
+                formik.setFieldValue("customTargetNetwork", "", false);
+              }
+              formik.validateForm();
+            }}
+            disabled={isEditing || !!editRestriction}
+            managedNetworks={networkList}
+            help={
+              formik.values.targetNetwork !== networkOtherLabel &&
+              "Peering is limited to intra-cluster OVN networks; non-OVN or cross-cluster network peering is not supported."
+            }
+            required
+          />
+        )
+      )}
+
+      {(formik.values.targetNetwork === networkOtherLabel ||
+        formik.values.targetProject === projectOtherLabel) && (
+        <Input
+          id="customTargetNetwork"
+          name="customTargetNetwork"
+          type="text"
+          label="Target network"
+          placeholder="Enter target network name"
+          value={formik.values.customTargetNetwork}
+          onChange={formik.handleChange}
+          onBlur={formik.handleBlur}
+          disabled={isEditing || !!editRestriction}
+          title={editRestriction}
+          help={
+            "Peering is limited to intra-cluster OVN networks; non-OVN or cross-cluster network peering is not supported."
+          }
+        />
+      )}
+
+      {formik.values.targetNetwork !== networkOtherLabel &&
+        formik.values.targetProject !== projectOtherLabel &&
+        !isEditing && (
+          <Input
+            id="createMutualPeering"
+            name="createMutualPeering"
+            type="checkbox"
+            label="Create mutual peering"
+            checked={formik.values.createMutualPeering}
+            onChange={(e) => {
+              formik.setFieldValue("createMutualPeering", e.target.checked);
+            }}
+            disabled={!!editRestriction}
+            title={editRestriction}
+          />
+        )}
```

Comment:
I think we should always display this input, but disable the field with a help text if either network or project are set to  `other`.